webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/administration/PropertiesManagement.scala

```diff
@@ -483,58 +485,186 @@ class PropertiesManagement extends DispatchSnippet with Loggable {
     ) apply (xml ++ Script(check()))
   }
 
-  def networkProtocolSection = { xml : NodeSeq =>
-    //  initial values, updated on successfull submit
-    def networkForm(initValue : SyslogProtocol) = {
-      var initReportsProtocol = initValue
-      var reportProtocol = initValue
-      def check = {
-        val noChange = initReportsProtocol == reportProtocol
-        Run(s"""$$("#networkProtocolSubmit").attr("disabled",${noChange});""")
+
+  def reportProtocolSection = { xml: NodeSeq =>
+    //  initial values, updated on successful submit
+    var initSyslogProtocol = configService.rudder_syslog_protocol.toBox
+    var initReportProtocol = configService.rudder_report_protocol_default.toBox
+    var initDisabledSyslog  = configService.rudder_syslog_protocol_disabled.toBox
+
+    // form values
+    var syslogProtocol = initSyslogProtocol.getOrElse(SyslogUDP)
+    var reportProtocol = initReportProtocol.getOrElse(AgentReportingHTTPS)
+    var disabledSyslog = initDisabledSyslog.getOrElse(true)
+
+    def noModif = (
+      for {
+        initSyslog <- initSyslogProtocol
+        initProtocol <- initReportProtocol
+        initDisabled <- initDisabledSyslog
+      } yield {
+        initSyslog == syslogProtocol &&
+          initProtocol == reportProtocol &&
+          initDisabled == disabledSyslog
+
       }
+      ).getOrElse(false)
 
-      def submit = {
-        val actor = CurrentUser.actor
-        configService.set_rudder_syslog_protocol(reportProtocol,actor,None).toBox match {
-          case Full(_) =>
-            // Update the initial value of the form
-            initReportsProtocol = reportProtocol
-            startNewPolicyGeneration
-            check & JsRaw("""createSuccessNotification("Network protocol options correctly updated")""")
-          case eb:EmptyBox =>
-            Noop & JsRaw("""createErrorNotification("Error when saving network protocol options")""")
+    def check() = {
+      Run( s""" $$("#reportProtocolSubmit").prop('disabled', ${noModif});""".stripMargin)
+    }
+
+    def submit = {
+      val actor = CurrentUser.actor
+      configService.set_rudder_report_protocol_default(reportProtocol).toBox.foreach(updateOk => initReportProtocol = Full(reportProtocol))
+      configService.set_rudder_syslog_protocol(syslogProtocol, actor, None).toBox.foreach(updateOk => initSyslogProtocol = Full(syslogProtocol))
+      configService.set_rudder_syslog_protocol_disabled(disabledSyslog, actor, None).toBox.foreach(updateOk => initDisabledSyslog = Full(disabledSyslog))
+
+      // start a promise generation, Since we check if there is change to save, if we got there it mean that we need to redeploy
+      startNewPolicyGeneration
```

Comment:
the configService should take care of starting the new policy generation if value changed (at least I think that's how it works)

Reply:
need to look at the service!